src/layout/header/HeaderLinks/HeaderLinks.tsx

```diff
@@ -1,42 +1,29 @@
-import { FC, ReactNode } from 'react'
+import { FC } from 'react'
 import { useNavigate } from 'react-router-dom'
 import { useTranslation } from 'react-i18next'
 
 import './HeaderLinks.scss'
 import { HEADER_LINKS } from 'src/routes'
 
-type ExternalLinkType = {
-  component: {
-    label: string
-    path: string
-    icon: ReactNode
-  }
-  t: (key: string) => string
-}
-
-type InternalLinkType = ExternalLinkType & {
-  component: ExternalLinkType['component'] & {
-    element: ReactNode
-  }
-}
+type LinkType = { component: (typeof HEADER_LINKS)[number] }
 
 const HeaderLinks: FC = () => {
-  const { t } = useTranslation()
   return (
     <div className="header-links">
       {HEADER_LINKS.map((item) => {
-        if (item.element == null) {
-          return <ExternalLink key={item.label} component={item} t={t} />
+        if (item.element === null) {
+          return <ExternalLink key={item.label} component={item} />
         } else {
-          return <InternalLink key={item.label} component={item} t={t} />
+          return <InternalLink key={item.label} component={item} />
         }
       })}
     </div>
   )
 }
 
-const ExternalLink: FC<ExternalLinkType> = ({ component, t }) => {
+const ExternalLink: FC<LinkType> = ({ component }) => {
```

Comment:
```suggestion
const ExternalLink = ({ label, path, icon }: typeof HEADER_LINKS)[number]) => {
```

Reply:
You can use prop spreading, but I think it's better to explicitly state what properties you want to send.
Consider, for example, a scenario where someone adds a `style` attribute to the `HEADER_LINKS`. It will cause the `ExternalLink` to get this styling and that would be quite confusing.
In general, [prop spreading isn't recommended](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-props-no-spreading.md) for most use cases and your design has similar effect